File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/Perm.java
package com.tencent.devops.scm.api.pojo;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 用户拥有的代码库权限
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Perm {
    private Boolean pull;
    private Boolean push;
    private Boolean admin;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/Perm.kt
package com.tencent.devops.scm.api.pojo

data class Perm(
    // 是否有拉取权限
    val pull: Boolean = false,
    // 是否有推送权限
    val push: Boolean = false,
    // 是否有管理员权限
    val admin: Boolean = false
)
